backend/pkg/api/data_access/vdb_management.go

```diff
@@ -836,83 +835,68 @@ func (d *DataAccessService) GetValidatorDashboardGroupExists(ctx context.Context
 }
 
 func (d *DataAccessService) AddValidatorDashboardValidators(ctx context.Context, dashboardId t.VDBIdPrimary, groupId uint64, validators []t.VDBValidator) ([]t.VDBPostValidatorsData, error) {
+	result := []t.VDBPostValidatorsData{}
+
 	if len(validators) == 0 {
 		// No validators to add
 		return nil, nil
 	}
 
-	pubkeys := []struct {
-		ValidatorIndex t.VDBValidator `db:"validatorindex"`
-		Pubkey         []byte         `db:"pubkey"`
-	}{}
-
-	addedValidators := []struct {
-		ValidatorIndex t.VDBValidator `db:"validator_index"`
-		GroupId        uint64         `db:"group_id"`
-	}{}
-
-	// Query to find the pubkey for each validator index
-	pubkeysQuery := `
-		SELECT
-			validatorindex,
-			pubkey
-		FROM validators
-		WHERE validatorindex = ANY($1)
-	`
-
-	// Query to add the validators to the dashboard and group
-	addValidatorsQuery := `
-		INSERT INTO users_val_dashboards_validators (dashboard_id, group_id, validator_index)
-			VALUES
-	`
-
-	for idx := range validators {
-		addValidatorsQuery += fmt.Sprintf("($1, $2, $%d), ", idx+3)
-	}
-	addValidatorsQuery = addValidatorsQuery[:len(addValidatorsQuery)-2] // remove trailing comma
-
-	// If a validator is already in the dashboard, update the group
-	// If the validator is already in that group nothing changes but we will include it in the result anyway
-	addValidatorsQuery += `
-		ON CONFLICT (dashboard_id, validator_index) DO UPDATE SET
-			dashboard_id = EXCLUDED.dashboard_id,
-			group_id = EXCLUDED.group_id,
-			validator_index = EXCLUDED.validator_index
-		RETURNING validator_index, group_id
-	`
-
-	// Find all the pubkeys
-	err := d.alloyReader.SelectContext(ctx, &pubkeys, pubkeysQuery, pq.Array(validators))
+	tx, err := d.userWriter.BeginTxx(ctx, nil)
```

Comment:
this should be `writerDb` not `userWriter`.

Reply:
Right, copy&paste error. Good find.